crates/energy/src/battery/mod.rs

```diff
@@ -0,0 +1,56 @@
+use bevy::prelude::*;
+
+pub(crate) mod component;
+mod systems;
+
+pub(crate) struct BatteryPlugin;
+
+impl Plugin for BatteryPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_system(systems::discharge_battery);
+    }
+}
+
+/// The rate at which the battery discharges in kiloJoules per second.
+const DISCHARGE_RATE: f32 = 30.;
+/// The minimum energy level in kiloJoules at which the unit can still move.
+const MIN_MOVE_ENERGY: f32 = 74.; // based on how many a car uses per second
+/// The minimum energy level in kiloJoules at which the unit can still attack.
+const MIN_ATTACK_ENERGY: f32 = 20_000.; // google said that a rail-gun takes 25 Mj
```

Comment:
There are two different concepts which are being misleadingly merge to a single one here:

* Minimum required energy for some kind of activity to keep happening. There are several reasons for this mechanism, the main one is that the vehicle does not loose basic functionality due to a death battery. Once below below e.g. 20% or some Joules, it is time to use energy wisely (i.e. movement only).
* Energy physically required to perform an action, e.g. charge needed to fire a laser gun.

Also, drones are currently equipped with lasers not rail-guns so the comment is a bit misleading. Not that the actual value matter much at this point during the development.

Reply:
I think that we should just use the system where battery is depleted by preforming different actions. These can be loaded from object config. Do you think I should just go ahead and make that happen? the other thing seems like another feature/PR (sounds like something that should be configurable though the UI E.g at 20% stop firing at low threat targets + go to nearest safe power source)